pkg/cmd/util/clientcmd/factory.go

```diff
@@ -53,83 +55,115 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 // Factory provides common options for OpenShift commands
 type Factory struct {
 	*kubecmd.Factory
-	OpenShiftClientConfig clientcmd.ClientConfig
+	OpenShiftClientConfig ClientConfig
+}
+
+// intercept errors that need special treatment for end-users (e.g. no config file present)
+type ClientConfig struct {
+	KClientConfig clientcmd.ClientConfig
+}
+
+func (cfg *ClientConfig) ClientConfig() (*kclient.Config, error) {
+	c, err := cfg.KClientConfig.ClientConfig()
+	err = WrapClientErrors(err)
```

Comment:
The main intent of this gymnastics around errors is to provide better messages for end-users, but it has to be done *everywhere*, not only in `login`. E.g., when you do `osc get pods` and a server could not be found by any mechanism we don't want "no server found", but [this](https://github.com/fabianofranz/origin/blob/osc_login_further/pkg/cmd/util/clientcmd/errors.go#L19). Same for certificate issues. 

Also, in `login` we need to be able to understand each one of them individually in order to be able to prompt for proper input in case they happen. 

So, I'm ok with refactorings around this (e.g. moving this `KClientConfig` out of the factory and handle it in `OpenShiftClientConfig` maybe), but it needs to apply to every client/config created for `osc`.